src/pyocr/builders.py

```diff
@@ -596,8 +612,8 @@ def write_file(file_descriptor, boxes):
     def start_line(self, box):
         pass
 
-    def add_word(self, word, box):
-        self.word_boxes.append(Box(word, box))
+    def add_word(self, word, box, confidence):
+        self.word_boxes.append(Box(word, box, confidence))
```

Comment:
`add_word()` functions are called by `pyocr.libtesseract.image_to_string()`. Since you haven't updated libtesseract support, this change will break it.

I suggest you try running the tests. Make sure you have `cuneiform`, `tesseract`, `libtesseract`, `tesseract-ocr-fra`, and `tesseract-ocr-jpn` installed, and simply run `./run_tests.py`.

Unfortunately, the output of the tests slightly vary based on the exact version of Tesseract and Cuneiform you're using (and wind direction I guess ....). So you will have to filter the failed tests manually: ignore those that failed just because the output has slightly changed, and just focus on the ones that failed because the API broke due to your changes.

Reply:
Fair point! Here is what I did:

- I've implemented the changes in `libtesseract/__init__.py` and `libtesseract/tesseract_raw.py` to support the extraction of the confidence measure using the C-API
- The confidence parameter for the Box constructor has a default value of 0

All in all, this basically means that the confidence measure is propagated to the output hocr files for the tesseract and libtesseract interfaces and a value of 0 is used for all the words when using cuneiform. Do you think that makes sense?

I've done some manual testing with Cuneiform, Tesseract and libtesseract to verify everything was working as expected (looking at the output hocr files). There are however still many failing unit-tests when running the test suite. I must say that it's quite hard to know if it's because I broke the API or just because tesseract feels like spitting out a different output 😅

Reply:
Yeah, I know, they are a pain. Unfortunately I haven't found a better way :/
Hint: if they fail on `assertEqual`, it's usually that the output differs from what was expected (aka the wind is blowing north now). For this change, you ignore those theses and focus on those where an uncatched Exception has been raised.